config_converter/config_mapper/config_mapper.py

```diff
@@ -48,21 +85,36 @@ def extract_root_dirs(config_obj: config_pb2.Directive) -> dict:
             sys.exit()
         plugin_name = plugin_prefix_map[d.name] + plugin_type
         if plugin_name not in _SUPPORTED_PLUGINS:
+            stats['entities_unrecognized'] += 1
+            stats['attributes_unrecognized'] +=\
+                _get_aggregated_num_attributes(d)
             print(f'We do not know plugin {plugin_name}')
         else:
             plugin_dir = dir_name_map[d.name]
             if plugin_dir not in logs_module:
                 logs_module[plugin_dir] = []
-            logs_module[plugin_dir].append(_convert_plugin(d, plugin_name))
+            current_attribute_count = stats['attributes_recognized']
+            logs_module[plugin_dir].append(
+                _convert_plugin(d, plugin_name, stats))
+            present_dir_attribute_count: int = _get_aggregated_num_attributes(
+                d)
+            if current_attribute_count + present_dir_attribute_count ==\
+                    stats['attributes_recognized']:
+                stats['entities_recognized_success'] += 1
+            elif current_attribute_count == stats['attributes_recognized']:
+                stats['entities_recognized_failure'] += 1
+            else:
+                stats['entities_recognized_partial'] += 1
             try:
                 result['logging_level'] = next(p.value for p in d.params
                                                if p.name == '@log_level')
+                stats['attributes_recognized'] += 1
             except StopIteration:
                 continue
-    return result
+    return (result, stats)
 
 
-def _convert_plugin(d: config_pb2.Directive, plugin: str) -> dict:
+def _convert_plugin(d: config_pb2.Directive, plugin: str, stats: dict) -> dict:
     """Cases on plugin, calls corresponding mapping function."""
```

Comment:
Let's update the docstring with more information, as this function interface is bigger now, and stats actually will be updated in-place - so it is worthy to let the reader know what to expect without understanding the implementation.

reference: https://google.github.io/styleguide/pyguide.html#doc-function-args

Reply:
You're right - I'll make it more thorough and follow the conventions mentioned in the link.